tools/link.py

```diff
@@ -789,6 +789,10 @@ def phase_linker_setup(options, state, newargs):
   if options.oformat in (OFormat.WASM, OFormat.BARE):
     # If the user asks directly for a wasm file then this *is* the target
     wasm_target = target
+  elif settings.SINGLE_FILE:
+    # In SINGLE_FILE mode the wasm file is not part of the output at all
+    # so we generate it the temp directory.
+    wasm_target = in_temp(shared.replace_suffix(target, '.wasm'))
```

Comment:
If we want to be this cautious about not-emitted files than wasm2js mode also does not emit a `.wasm`.

Reply:
Good point. Done.